File: Szakdoga.UI/Angular/src/app/shared/models/syllabus.ts
import {Subject} from "./subject";

export  class Syllabus{
  id:string;
  name:string;
  length:number;
  mustChoseCredit:number;
  chosableCredit:number;
  startingSpecSemester:number;
  parent:string;
  subjects:Subject[];


  constructor(prop:any) {
    this.id = prop.id;
    this.name = prop.name;
    this.length = +prop.length;
    this.mustChoseCredit = +prop.mustChoseCredit;
    this.chosableCredit = +prop.chosableCredit;
    this.startingSpecSemester = +prop.startingSpecSemester;
    this.parent = prop.parent;
    this.subjects = prop.subjects;
  }
}

File: Szakdoga.DataLayer/Model/Syllabus.cs
namespace Szakdoga.DataLayer.Model
{
    public partial class Syllabus
    {
        public Syllabus()
        {
            InverseParentNavigation = new HashSet<Syllabus>();
            Subjects = new HashSet<Subject>();
            Students = new HashSet<Student>();
        }

        public string Id { get; set; } = null!;
        public string? Name { get; set; }
        public string? Length { get; set; }
        public string? MustChoseCredit { get; set; }
        public string? ChosableCredit { get; set; }
        public string? StartingSpecSemester { get; set; }
        public string? Parent { get; set; }

        public virtual Syllabus? ParentNavigation { get; set; }
        public virtual ICollection<Syllabus> InverseParentNavigation { get; set; }
        public virtual ICollection<Subject> Subjects { get; set; }

        public virtual ICollection<Student> Students { get; set; }
    }
}
